src/reducers/select.js

```diff
@@ -47,65 +48,91 @@ export default (pattern) => (base, source) => {
+        debug(`All subfields are equal`);
+        debug(`base is: ${JSON.stringify(base, undefined, 2)}`);
+        return base;
+        // Might just as well return source here, since source and base are identical?
+        // Then the same equality and subset comparison logic could be used regardless of whether source and base have the same number of subfields or source has more
+        // I.e. just one if block: if (sourceSubsNormalized.length >= baseSubsNormalized.length) (it works at least for these test cases)
+      }
+      // Then compare subsets of subfield values
+      if (strictEqualityComparison.some(element => element === false)) {
+        debug(`in subset comparison`);
+        const subsetComparison = baseSubsNormalized.map(compareSubsets);
+        debug(`subsetComparison: ${JSON.stringify(subsetComparison, undefined, 2)}`);
+        if (subsetComparison.every(element => element === true)) {
+          debug(`All base subfields are subsets of source subfields`);
+          return source;
+        }
+      }
+    }
 
     // If source has more subfields than base
-    if (sourceSubs.length > baseSubs.length) {
-      // And if strict equality is true for all base subfields
-      // The result array is filled with compareStrictEquality return values
-      // And if they are all true, it means that source has equal subfields to all base subfields
-      const result = baseSubs.map(compareStrictEquality);
-      debug(`result: ${JSON.stringify(result, undefined, 2)}`);
-      if (result.every(element => element === true)) {
-        // Source is returned since it contains all of base and more
+    if (sourceSubsNormalized.length > baseSubsNormalized.length) {
+      // First compare strict equality (default)
+      const strictEqualityLongerSource = baseSubsNormalized.map(compareStrictEquality);
+      debug(`strictEqualityLongerSource: ${JSON.stringify(strictEqualityLongerSource, undefined, 2)}`);
+      if (strictEqualityLongerSource.every(element => element === true)) {
         debug(`Source is a superset of base`);
+        debug(`source is: ${JSON.stringify(source, undefined, 2)}`);
         return source;
       }
-      // If source is not a proper superset of base, return the original unmodified base
-      // In this case the result array contains at least one element that is false
-      debug(`Source is NOT a superset of base`);
-      return base;
+      // Then compare subsets of subfield values
+      if (strictEqualityLongerSource.some(element => element === false)) {
+        debug(`in subset comparison`);
+        const subsetLongerSource = baseSubsNormalized.map(compareSubsets);
+        debug(`subsetLongerSource: ${JSON.stringify(subsetLongerSource, undefined, 2)}`);
+        if (subsetLongerSource.every(element => element === true)) {
+          debug(`All base subfields are subsets of source subfields`);
+          return source;
+        }
+      }
     }
```

Comment:
At this point, filter out the subfields from each field which have a equal counterpart in the opposite field (Using the equality function. Then use those arrays in further comparisons.